### DISCUSSION TITLE
Tags/Labels feature

### DISCUSSION THREAD

**Discussion by lilaelephant** [dani-garcia/vaultwarden#2333] [Ideas]
This isn't a new feature request and many people have been asking for this feature for a long time ([5 years](https://github.com/bitwarden/server/issues/9)!!). I'm starting this discussion as my motivation is personal and strategical _(see section "What I am hoping for")_.

### Feature request background

I have been following the community and github discussions around this topic since day 0 and I'm disappointed that this feature is still not part of Bitwarden nor in any roadmap.

There are many use-cases _(see [community](https://community.bitwarden.com/t/tagging-items-in-vault/132))_ but the one that I want to highlight is the possibility to answer questions like those:

* Who has my government id? (tagging items with #gov-id)
* I just moved and my address changed, where do I need to change my address to the new one? (tagging items with #address/street-name-75)
* My phone number changed, where do I need to change it, so I'm not getting locked out of accounts? (tagging items with #phone-numbers/***7523)
* credit card expired, ...
* who knows my real name, date of birth, ...

But from what I see, there are no plans to add this to Bitwarden. As this is the missing feature that I need to have in order to migrate away from 1Password, **I am ready to help with the development**.

### What I am hoping for

I would love to see this feature in Vaultwarden and it's web build. I think the majority of users who need this feature wouldn't use it on their mobile devices, it would be sufficient to have this feature in the web version.

I understand that Vaultwarden is a rust server implementation of Bitwarden which means that it is likely that this feature would never make it in any official client implementation besides the vaultwarden UI. And this is where things could become a bit controversial as this would need patching work in the https://github.com/dani-garcia/bw_web_builds repo.

As Vaultwarden is a community project (thanks @dani-garcia ❤️) I wanted to open this discussion to ask if this is something the community would like to support. Also, as this server implementation has a massive user base, I could imagine it being an opportunity to eventually get this implemented in the main projects.

**Comment by BlackDex** [comment]
It sound like a useful feature. The only thing is, we want to make as less as possible changes too the web-vault, and also try to keep as close to the Bitwarden project as possible.
Adding a feature like this would be huge and with every new release of Bitwarden Web we would need to patch that all over again which is very time consuming.

Also, as you mentioned it your self, this is a Rust project, and i don't know about the other maintainers, but my TypeScript skills are limited.

I think it would be more beneficial if someone who is capable of adding something like this, that they add this to the upstream project and not only for Vaultwarden.

**Reply by lilaelephant** [reply]
Thanks so much for your reply!

Do you think it would be feasible to implement this feature in the official web-vault without having it implemented in the official server project? I'm comfortable with TypeScript and I might be able to gather some people who could support me/us with that.

C# isn't in my circle of competence, so that would be a problem. But maybe the web-vault could check for some kind of feature flag/endpoint to see if tags/labels are supported.

**Comment by macross5** [comment]
+1 
I would like that feature aswell. I need a feature to give a entry multiple "names" to expand the search results. I think that I have to workaround it with a complex entry name.

**Comment by nv1t** [comment]
From my perspective, there is less change if you use collections for this feature. we have 100+ Collections and pretty much use them as labels/tags.

we would love to have this in the web client as well. this would make it even more usable.

**Comment by BlackDex** [comment]
I just want to make sure again. Those feature request needs to be requested at Bitwarden. Or, if someone can code it them self could try to create a PR.

It doesn't help to +1 or confirm it would be useful here.
